src/main/java/org/springframework/data/elasticsearch/core/DocumentOperations.java

```diff
@@ -158,6 +186,34 @@ default void bulkUpdate(List<UpdateQuery> queries, IndexCoordinates index) {
 	 */
 	String delete(String id, IndexCoordinates index);
 
+	/**
+	 * Delete the one object with provided id.
+	 *
+	 * @param id the document ot delete
+	 * @param entityType must not be {@literal null}.
+	 * @return documentId of the document deleted
+	 */
+	String delete(String id, Class<?> entityType);
+
+	/**
+	 * Deletes the given entity
+	 *
+	 * @param entity the entity to delete
+	 * @param <T> the entity class
+	 * @return documentId of the document deleted
+	 */
+	<T> String delete(T entity);
```

Comment:
Nit: `<T>` is not required as we do not make any type contracts here. `delete(Object)` is fine.